ckan/cli/__init__.py

```diff
@@ -34,45 +34,63 @@ def _read_config_file(self, filename):
+
+    def _unwrap_config_chain(self, filename):
+        """Get all names of files in use-chain.
+
+        Parse files using RawConfigParser, because top-level config file can
+        use variaables from the lower-level config files, which are not
+        initialized yet.
+        """
+        parser = RawConfigParser()
+        chain = []
+        while True:
+            parser.read(filename)
+            chain.append(filename)
+            use = parser.get(self.section, "use")
+            if not use:
+                return chain
+            try:
+                schema, next_config = use.split(":", 1)
+            except ValueError:
+                raise CkanConfigurationException(
+                    "Missing colon symbol in the value of `use` " +
+                    f"option inside {filename}: {use}"
+                )
+
+            if schema != "config":
+                return chain
+            filename = os.path.join(
+                os.path.dirname(os.path.abspath(filename)), next_config)
+            if filename in chain:
+                joined_chain = ' -> '.join(chain + [filename])
+                raise CkanConfigurationException(
+                    'Circular dependency located in '
+                    f'the configuration chain: {joined_chain}'
+                )
 
     def _create_config_object(self):
-        use_config_path = self.config_file
-        self._read_config_file(use_config_path)
-
-        # # The global_config key is to keep compatibility with Pylons.
-        # # It can be safely removed when the Flask migration is completed.
-        self.config[u'global_conf'] = self.parser.defaults().copy()
-
-        self._update_config()
-
-        loaded_files = [use_config_path]
-
-        while True:
-            schema, path = self.parser.get(self.section, u'use').split(u':')
-            if schema == u'config':
-                use_config_path = os.path.join(
-                    os.path.dirname(os.path.abspath(use_config_path)), path)
-                # Avoid circular references
-                if use_config_path in loaded_files:
-                    chain = ' -> '.join(loaded_files + [use_config_path])
-                    raise CkanConfigurationException(
-                        'Circular dependency located in '
-                        f'the configuration chain: {chain}'
-                    )
-                loaded_files.append(use_config_path)
-
-                self._read_config_file(use_config_path)
-                self._update_config()
-            else:
-                break
+        chain = self._unwrap_config_chain(self.config_file)
+        for filename in reversed(chain):
+            self._read_config_file(filename)
+            # The global_config key is to keep compatibility with Pylons.  It
+            # can be safely removed when the Flask migration is completed.
+            self.config[u'global_conf'] = self.parser.defaults().copy()
```

Comment:
I think we can safely remove this since we don't need to maintain compatibility with Pylons anymore.